src/main/java/com/es/phoneshop/web/CheckoutPageServlet.java

```diff
@@ -0,0 +1,116 @@
+        request.setAttribute("order", orderService.getOrder(cart));
+        request.setAttribute("paymentMethods", PaymentMethod.values());
+        request.getRequestDispatcher("/WEB-INF/pages/main/checkout.jsp").forward(request, response);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        Cart cart = cartService.getCart(new HttpSessionCartLoader(request));
+        Order order = orderService.getOrder(cart);
+        Map<String, String> errors = new HashMap<>();
+        setRequiredParameter(request, "firstName", errors, order::setFirstName);
+        setRequiredParameter(request, "lastName", errors, order::setLastName);
+        setPhone(request, errors, order);
+        setDeliveryDate(request, errors, order);
+        setRequiredParameter(request, "deliveryAddress", errors, order::setDeliveryAddress);
+        setPaymentMethod(request, errors, order);
+        if (errors.isEmpty()) {
+            orderService.placeOrder(order);
+            cartService.clearCart(cart);
+            response.sendRedirect(request.getContextPath() + "/order/overview/" + order.getSecureId());
+        } else {
+            request.setAttribute("errors", errors);
+            request.setAttribute("order", order);
+            request.setAttribute("paymentMethods", orderService.getPaymentMethod());
+            request.getRequestDispatcher("/WEB-INF/pages/main/checkout.jsp").forward(request, response);
+        }
+    }
+
+    private void setRequiredParameter(HttpServletRequest request, String parameter,
+                                      Map<String, String> errors, Consumer<String> consumer) {
+        String value = request.getParameter(parameter);
+        if (value == null || value.isEmpty()) {
+            errors.put(parameter, "Value is required");
+        } else {
+            consumer.accept(value);
+        }
+    }
+
+
+    private void setPhone(HttpServletRequest request, Map<String, String> errors, Order order) {
+        String value = request.getParameter("phone");
+        if (value == null || value.isEmpty()) {
+            errors.put("phone", "Value is required");
+        } else if (!value.matches("\\+?\\d*")) {
+            errors.put("phone", "Invalid value");
+        } else {
+            order.setPhone(value);
+        }
+    }
+
+    private void setDeliveryDate(HttpServletRequest request, Map<String, String> errors, Order order) {
+        String value = request.getParameter("deliveryDate");
+        if (value == null || value.isEmpty()) {
+            errors.put("deliveryDate", "Value is required");
+        } else {
+            try {
+                DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+                LocalDate date = LocalDate.parse(value, formatter);
+                if (date.compareTo(LocalDate.now()) <= 0) {
+                    throw new PastDateException();
+                }
```

Comment:
Throw exception is unnecessary here, since you catch it right below.
And actually no need to change `compareTo` here (despite what i said on the call), just take look at implementation of LocalDate.compareTo(..) and LocalDate.isBefore(..)